src/check_datapackage/check.py

```diff
@@ -304,6 +306,64 @@ def _handle_S_resources_x_schema_fields_x(
     return edits
 
 
+def _handle_S_resources_x_schema_fields_x_constraints_enum(
+    parent_error: SchemaError,
+    schema_errors: list[SchemaError],
+) -> SchemaErrorEdits:
+    """Only flag errors for the relevant field type and simplify errors."""
+    edits = SchemaErrorEdits(remove=[parent_error])
+    errors_in_group = _get_errors_in_group(schema_errors, parent_error)
+
+    # Remove errors for other field types
+    if _error_not_for_field_type(parent_error):
```

Comment:
```suggestion
    if _not_field_type_error(parent_error):
```

Sounds a bit better like this.